src/support.js

```diff
@@ -575,6 +575,13 @@ function loadWebAssemblyModule(binary, flags) {
         assert(table.get(tableBase + i) !== undefined, 'table entry was not filled in');
       }
 #endif
+      if (!functionsInTableMap) var functionsInTableMap = new WeakMap();
```

Comment:
the `var` here is incorrect - it will leave `functionsInTableMap` as a local variable in this function.

how about adding a `populateTableMap(start, end)` function in `runtime_functions.js`, which adds functions in the range `[start, end)`? Then the main module would call it with `0, table.length` and here we could call it with `tableBase, table.length`.

Reply:
I just realized there is no need to create `functionsInTableMap` here.   If it hasn't been created yet we can just do nothing and let it be created on first use of `addFunction`.  We only need to take action here if it has already been created since it will be stale.  If it doesn't exist yet its not stale.